app/serializers/export/contact_serializer.rb

```diff
@@ -1,12 +1,28 @@
 module Export
   class ContactSerializer < ActiveModel::Serializer
     attributes :sharing,
-               :receiving,
+               :following,
```

Comment:
No, when you can follow without sharing (for example follow a newsfeed, but don't want to share the profile to the feed), then you need the information federated. So when `following` in the `contact` message is true, you need to send all (public) posts to this user. That's why we currently federate always true for `following`.

So in the export we need `sharing` and `following` for the users setting of the contact, and `receiving` and `subscribed` for if we are allowed to display the profile, and send private messages, and if we need to send new posts to that contact.

So we should export `sharing` and `receiving` from the database, and export `following` with the same value as `sharing` and `subscribed` with the same value as `receiving` (instead of always `true` what I wrote above, because it's wrong). That represents the way how diaspora currently works and also what we currently federate with the `contact` message.